src/Renderer/Shader/TileFS.glsl

```diff
@@ -122,21 +122,22 @@ void main() {
                             textureIndex,
                             projWGS84 ? vUv_WGS84 : uvPM);
 
-                        if (layerColor.a > 0.0) {
+                        // (!validTexture) test : force to have a white background for mixing color layers
```

Comment:
IMHO, instead of adding more ad-hoc code to TileFS (ie: customized rendering based on a float value in paramsA) you should instead expose a feature allowing the user to insert their custom glsl code here to modify diffuseColor for a given layer.

Reply:
it's an existing code fix not an improvement.
it takes another PR

Reply:
Hmmm I'm pretty sure you're adding new code ( `} else if(paramsA.z > 0.0) {` ...)...

Reply:
No, I do not think so

Reply:
Ok. I realized that you duplicated the code that was shared before.
I don't understand your comments :
- "// separation of states to avoid errors in final statement" ?
- "// (!validTexture) test : force to have a white background for mixing color layers" ?

Reply:
I remove `// separation of states to avoid errors in final statement`

and change test `(!validTexture)` to `(layerColor.a == 0.0 && !validTexture)` because if first texture is transparent, we need to force blending with CWhite

Reply:
> I remove // separation of states to avoid errors in final statement

Then please explain why you need to duplicate the code logic.